src/polygon.c

```diff
@@ -60,7 +60,7 @@ PG_FUNCTION_INFO_V1(spheretrans_poly_inverse);
 PG_FUNCTION_INFO_V1(spherepoly_add_point);
 PG_FUNCTION_INFO_V1(spherepoly_add_points_finalize);
 PG_FUNCTION_INFO_V1(spherepoly_is_convex);
-
+PG_FUNCTION_INFO_V1(spherepoly_from_point_array);
```

Comment:
Just wondering why you didn't add this new function to polygon.h? I did in PR #96 and am wondering if I shouldn't have?

Reply:
@esabol Function declaration in a header is needed when you use this function in another translation unit (.c file). You just include this header with function declarations to be able to use these functions without any compiler warnings. The spherepoly_from_point_array function is an interface function that is searched (ldsym) and called by postgresql. It is not used in other .c sources. Once, some other functions are defined in .h file, I will add the declaration. 

Furthermore, pgindent requires "extern" modificator for functions declarations. I have some plans to use pgindent to reformat .h files to satisfy the postgresql coding style.

Reply:
@vitcpp wrote:
> Function declaration in a header is needed when you use this function in another translation unit (.c file).

Uh, yeah. I've been a professional C developer for over 28 years. I know what `.h` files are used for. 

In _my_ C projects, I put the prototype for _every_ function in its corresponding `.h` file. The only times I don't are when I'm working on an API library and I need to be concerned about which functions are in the public API and which are meant only to be used internally. And even then I'd probably create a separate internal-only `.h` file for the prototypes of internal functions. Anyway, that's just my development philosophy, I guess. Apparently, you have a different philosophy, and that's fine.

Basically, I just want to know if I should remove the prototype for `spherepoly_deg` that I added to `src/polygon.h` in PR #96. It sounds like I should!

Reply:
@esabol I'm sorry, I have no any intentions to teach anyone. I just wanted to clearly describe my position. My bad, a bucket with apples from me! :)

I agree with you to put the function declaration into .h file in pgsphere to follow the pgsphere code style. All other API functions are declared in .h files. I would also propose to add extern, but not in this PR. Pgindent will properly apply formatting to function declarations if it sees the extern keyword.

I will rebase and update my PR in according with your comments after the merge of #96.

Reply:
OK, cool.